testserver: populate startSingleNode param of RunInitialSQL correctly

rafiss opened the pull request:
RunInitialSQL accepts a flag that disables all replication in the cluster if it is true. This should be set for testservers that are a single node cluster.

Before this change, only `testserver.Cfg.DefaultZoneConfig.NumReplicas` would be set to 1, but not any of the other builtin ranges, like the meta or system ranges.

Epic: None
Release note: None

rafiss committed:
testserver: populate startSingleNode param of RunInitialSQL correctly

RunInitialSQL accepts a flag that disables all replication in the
cluster if it is true. This should be set for testservers that are a
single node cluster.

Before this change, only `testserver.Cfg.DefaultZoneConfig.NumReplicas`
would be set to 1, but not any of the other builtin ranges, like the
meta or system ranges.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110236)
<!-- Reviewable:end -->

knz reviewed (approved):
nice.

rafiss committed:
*: move call to yaml.FutureLineWrap() to happen during init

Previously this function which had global side-effects would only be
called if somebody happened to alter zone configs.

Now, it is applied universally, which makes test output more consistent.

Release note: None

rafiss commented:
tftr! i added a second commit which changes how yaml's are dumped out in tests. previously this would happen only if the test happened to alter zone configs, but now it happens in an `init()`. i can backport this change, if desired, after a few days.

bors r=knz
